modP2pImpl/src/org/aion/p2p/impl/comm/NodeMgr.java

```diff
@@ -158,9 +155,19 @@ public boolean isSeedIp(String _ip) {
      * @param _n Node
      */
     @Override
-    public synchronized void addTempNode(final INode _n) {
-        if (tempNodes.size() < maxTempNodes) {
-            tempNodes.add(_n);
+    public void addTempNode(final INode _n) {
+        final ReentrantLock putLock = this.putLock;
+        try {
+            putLock.lockInterruptibly();
+
+            if (tempNodes.size() < maxTempNodes && !tempNodes
+                .containsKey(_n.getPeerId()) && (notActiveNode(_n.getIdHash()) || _n.getIfFromBootList())) {
```

Comment:
_n.getPeerId() is 0 if during Node's constructor, the if-check on line 94 of Node.java didn't succeed.  Does this logic still work if that happens?

Reply:
the constructor is not coding neat, but the usage classes do the format check. Therefore the logic is always correct so far